src/arkode/arkode_splittingstep.c

```diff
@@ -403,88 +389,144 @@ static void splittingStep_PrintMem(const ARKodeMem ark_mem, FILE* const outfile)
-  Resets all SplittingStep optional inputs to their default
-  values. Does not change problem-defining function pointers or
-  user_data pointer.
-  ---------------------------------------------------------------*/
-static int splittingStep_SetDefaults(const ARKodeMem ark_mem)
+/*------------------------------------------------------------------------------
+  Resets all SplittingStep optional inputs to their default values. Does not
+  change problem-defining function pointers or user_data pointer.
+  ----------------------------------------------------------------------------*/
+static int splittingStep_SetDefaults(ARKodeMem ark_mem)
 {
   ARKodeSplittingStepMem step_mem = NULL;
   int retval = splittingStep_AccessStepMem(ark_mem, __func__, &step_mem);
   if (retval != ARK_SUCCESS) { return retval; }
 
-  retval = splittingStep_SetOrder(ark_mem, 0);
-  if (retval != ARK_SUCCESS) { return retval; }
-
-  /* TODO(SBR): This may cause issues if a user calls ARKodeSetDefaults. This
-   * issues affects other ARKODE steppers as well */
-  ARKodeSetInterpolantType(ark_mem, ARK_INTERP_LAGRANGE);
+  return splittingStep_SetOrder(ark_mem, 0);
+}
 
-  return ARK_SUCCESS;
+/*------------------------------------------------------------------------------
+  This routine checks if all required vector operations are present. If any of
+  them is missing it returns SUNFALSE.
+  ----------------------------------------------------------------------------*/
+static sunbooleantype splittingStep_CheckNVector(N_Vector y)
+{
+  return y->ops->nvlinearsum != NULL && y->ops->nvscale != NULL;
 }
 
-/*---------------------------------------------------------------
-  Creates the SplittingStep integrator
-  ---------------------------------------------------------------*/
-void* SplittingStepCreate(SUNStepper* const steppers, const int partitions,
-                          const sunrealtype t0, const N_Vector y0,
-                          const SUNContext sunctx)
+/*------------------------------------------------------------------------------
+  This routine validates arguments when (re)initializing a SplittingStep
+  integrator
+  ----------------------------------------------------------------------------*/
+static int splittingStep_CheckArgs(ARKodeMem ark_mem, SUNStepper* steppers,
+                                   int partitions, N_Vector y0)
 {
   if (steppers == NULL)
   {
-    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
                     "steppers = NULL illegal.");
-    return NULL;
+    return ARK_ILL_INPUT;
+  }
+
+  if (partitions <= 1)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "The number of partitions must be greater than one");
+    return ARK_ILL_INPUT;
   }
 
   for (int i = 0; i < partitions; i++)
   {
     if (steppers[i] == NULL)
```

Comment:
As with ForcingStep, the documentation for SplittingStep states that the provided steppers must implement all but one operation.  Those required operations should be checked here inside the loop.